model/riscv_insts_vext_utils.sail

```diff
@@ -0,0 +1,703 @@
+      /* Mask tail is always agnostic */
+      //result[i] = bitone
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    }
+    else {
+      /* Active body elements */
+      mask_helper[i] = true
+    }
+  };
+
+  (result, mask_helper)
+}
+
+/* Mask handling for cmp functions that use masks as output */
+val init_masked_result_cmp : forall 'n 'm, 8 <= 'm <= 128. (int('n), int('m), real, vector('n, dec, bool), vector('n, dec, bool)) -> (vector('n, dec, bool), vector('n, dec, bool)) effect {escape, rreg, undef, wreg}
+function init_masked_result_cmp(num_elem, vsew_bits, lmul, vd_val, vm_val) = {
+  let start_element : int                       = get_start_element();
+  let end_element   : int                       = get_end_element();
+  let mask_ag       : agtype                    = get_vtype_vma();
+  mask_helper       : vector('n, dec, bool)     = undefined;
+  result            : vector('n, dec, bool)     = undefined;
+
+  /* Determine the actual number of elements when lmul < 1 */
+  var real_num_elem : int = undefined;
+  if lmul >= 1.0 then {
+    real_num_elem = num_elem
+  }
+  else {
+    real_num_elem = floor(lmul * to_real(num_elem))
+  };
+  assert(num_elem >= real_num_elem);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if i < start_element then {
+      /* Prestart elements defined by vstart */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    }
+    else if i > end_element then {
+      /* Tail elements defined by vl */
+      /* Mask tail is always agnostic */
+      //result[i] = bitone
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    }
+    else if i >= real_num_elem then {
+      /* Tail elements defined by lmul < 1 */
+      /* Mask tail is always agnostic */
+      //result[i] = bitone
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    }
+    else if vm_val[i] == false then {
+      /* Inactive body elements defined by vm */
+      if mask_ag == UNDISTURBED then {
+        result[i] = vd_val[i]
+      }
+      else if mask_ag == AGNOSTIC then {
+        //result[i] = bitone
+	      result[i] = vd_val[i]
+      };
+      mask_helper[i] = false
+    }
+    else {
+      /* Active body elements */
+      mask_helper[i] = true
+    }
+  };
+
+  (result, mask_helper)
+}
+
+/* Misc number functions */
+val is_zeros : forall 'n, 'n >= 0. bits('n) -> bool
+function is_zeros(bitval) = {
+  bitval == zeros('n)
+}
+
+val is_ones : forall 'n, 'n >= 0. bits('n) -> bool
+function is_ones(bitval) = {
+  bitval == ones('n)
+}
+
+/* Floating point canonical NaN for 16-bit, 32-bit, 64-bit and 128-bit types */
+val canonical_NaN : forall 'm, 'm in {16, 32, 64, 128}. int('m) -> bits('m)
+function canonical_NaN('m) = {
+  match 'm {
+    16   => 0x_7e00,
+    32   => 0x_7fc0_0000,
+    64   => 0x_7ff8_0000_0000_0000,
+    128  => 0x_7fff_8000_0000_0000_0000_0000_0000_0000
+  }
+}
+
+/* Floating point NaN boxing / unboxing that support 16-bit to 128-bit types */
```

Comment:
These likely belong in F files, not V ones? They duplicate code that's already there...

Reply:
F files have separate functions for 32-bit and 64-bit NaN boxing, and do not support 16 or 128 bit, so we add a new function for the convenience of vector floating point instructions.

Reply:
1. They do support 16-bit for Zfh
2. They will need to support 128-bit for Q
3. Even if they didn't, the code should be extended and shared, not duplicated

This whole set of float functions belongs in common code.

Reply:
Currently Q  is not ratified. It's a support function, so put this into common, after this PR is merged into vector-dev